### PR TITLE
Introduced input trigger for date-picker

### PR DESCRIPTION
**Pull request by phytonmk**
## Description

Introduced input trigger for date-picker. In general it works but there are still some doubts. Also I need to update documentation, handle button trigger deprecation in better way and add tests.

Old button trigger is still used by default for backward compatibility. Hope we will replace it in nect major release. 

## Motivation and Context

Button trigger is inaccessible for blind users and not handy for keyboard only users. My hypothesis is that input trigger will resolve these issues.  

## How has this been tested?

I have added more vo-tests

## Screenshots:

Screenshot 2022-10-07 at 19 40 28

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by freeyoungstrong — APPROVED** [review]
LGTM. just some problems with tests and lint

File: semcore/date-picker/src/MonthPicker.jsx [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import React from 'react';
 import dayjs from 'dayjs';
 import createComponent from '@semcore/core';
-import { Header, Next, Popper, Prev, Title, Trigger } from './components';
+import { Header, Next, Popper, Prev, Title, Trigger, InputTrigger } from './components';
 import { CalendarMonths as Calendar } from './components/Calendar';
 import PickerAbstract from './components/PickerAbstract';
 
+const inputTriggerParts = { day: false, month: true, year: true };
```
Comment (ekrasnopolskaia) [inline semcore/date-picker/src/MonthPicker.jsx:8]: but these are not parts of the trigger, they are parts of the date

File: semcore/date-picker/src/MonthRangePicker.jsx [hunk 2]
```diff
@@ -2,10 +2,12 @@ import React from 'react';
 import dayjs from 'dayjs';
 import createComponent from '@semcore/core';
 import shortDateRangeFormat from './utils/shortDateRangeFormat';
-import { Header, Next, Period, Popper, Prev, Title, Trigger } from './components';
+import { Header, Next, Period, Popper, Prev, Title, Trigger, InputTrigger } from './components';
 import { CalendarMonths as Calendar } from './components/Calendar';
 import RangePickerAbstract, { Apply, Reset } from './components/RangePickerAbstract';
 
+const inputTriggerParts = { day: false, month: true, year: true };
```
Comment (ekrasnopolskaia) [inline semcore/date-picker/src/MonthRangePicker.jsx:9]: why there is a day

File: semcore/date-picker/src/components/Calendar.jsx [hunk 3]
```diff
@@ -122,6 +122,7 @@ class CalendarAbstract extends Component {
 
         fire(this, 'onHighlightedChange', highlighted);
         fire(this, 'onChange', date.toDate());
+        // fire(this, 'onVisibleChange', false);
```
Comment (ekrasnopolskaia) [inline semcore/date-picker/src/components/Calendar.jsx:125]: forgot the comments

File: semcore/date-picker/src/components/InputTrigger.jsx [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  Root,
+  ...otherProps
+}) => {
+  const { sep, order } = React.useMemo(() => {
+    const exampleDate = new Date(2000, 4, 29);
+    const options = { year: 'numeric', month: '2-digit', day: '2-digit' };
+    const dateTimeFormat = new Intl.DateTimeFormat(locale, options);
+
+    let sep = undefined;
```
Comment (ekrasnopolskaia) [inline semcore/date-picker/src/components/InputTrigger.jsx:252]: what does "sep" mean?

**Review by ekrasnopolskaia — COMMENTED** [review]
- Why are they so far away from each other? 
 Screenshot 2022-10-14 at 16 46 01
- I also assume that MonthRangePicker shouldn't have days like in MonthPicker 
 Screenshot 2022-10-14 at 16 48 33
- moves out a little bit

Screenshot 2022-10-14 at 16 54 43
- some problem with Custom header

https://user-images.githubusercontent.com/15107335/195879081-2c3ec91b-d675-4aa0-9552-ac74cadc38f0.mov


- Week picker shows the second date as disabled
Screenshot 2022-10-14 at 16 59 04

-icon color is differs from Figma

File: semcore/date-picker/src/MonthRangePicker.jsx (hunk 2 above)
Comment (phytonmk) [reply semcore/date-picker/src/MonthRangePicker.jsx:9]: Why not?

File: semcore/date-picker/src/components/InputTrigger.jsx (hunk 4 above)
Comment (phytonmk) [reply semcore/date-picker/src/components/InputTrigger.jsx:252]: It is common shortening for the separator word. Like the one exported from std lib `path` package

**Comment by lsroman** [comment]
Lena described all the main points 👌 I think, we need to synchronize with designers about the size and jumps of the font.
I also wanted to add that the up and down arrows should work.

File: semcore/date-picker/src/MonthRangePicker.jsx (hunk 2 above)
Comment (phytonmk) [reply semcore/date-picker/src/MonthRangePicker.jsx:9]: Days are disabled (`day: false`) 🙂

Comment (ekrasnopolskaia) [reply semcore/date-picker/src/MonthRangePicker.jsx:9]: But I can see them in the screenshot above

File: semcore/input-mask/CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,16 @@
 
 CHANGELOG.md standards are inspired by [keepachangelog.com](https://keepachangelog.com/en/1.0.0/).
 
+## [4.3.3] - 2022-10-30
```
Comment (ekrasnopolskaia) [inline semcore/input-mask/CHANGELOG.md:5]: future?

**Review by ekrasnopolskaia — COMMENTED** [review]
I still think the indents is too big. And is it needed to check the validity of the period entered from the keyboard?
Screenshot 2022-10-27 at 14 28 30
It might also be nice to find out if it needs to switch to text input on the hover and align it if so

File: semcore/input-mask/CHANGELOG.md (hunk 5 above)
Comment (phytonmk) [reply semcore/input-mask/CHANGELOG.md:5]: Planned rolling out date

**Pull request merged**